third-party/linshare/src/main/java/org/apache/james/linshare/client/GrantUserPermissionRequest.java

```diff
@@ -0,0 +1,46 @@
+package org.apache.james.linshare.client;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.util.List;
+import java.util.Objects;
+
+public class GrantUserPermissionRequest {
+    private final String uuid;
+    private final String mail;
+    private final String name;
+    private final List<String> permissions;
+
+    @VisibleForTesting
+    public GrantUserPermissionRequest(String uuid,
+                                      String mail,
+                                      String name,
+                                      List<String> permissions) {
+        this.uuid = uuid;
+        this.mail = mail;
+        this.name = name;
+        this.permissions = permissions;
+    }
+
+    public String getUuid() { return uuid; }
+    public String getMail() { return mail; }
+    public String getName() { return name; }
+    public List<String> getPermissions() { return permissions; }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```

Comment:
Again we have a common POJO template for equals and hashcode. Avoiding tiny differences here makes the review easier